### ISSUE TITLE
AZCOPY_MSI_RESOURCE_STRING does not work with MSAL

### ISSUE THREAD

**Issue by wanzhuM** [Azure/azure-storage-azcopy#2395]
### Which version of the AzCopy was used? 
v10.21.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux (specifically shell extension in ev2)

### What command did you run?
```
export AZCOPY_AUTO_LOGIN_TYPE=MSI
export AZCOPY_MSI_RESOURCE_STRING="$resourceId"
azcopy sync "$src" "$dst" --recursive --delete-destination=true
```

### What problem was encountered?
```
2023-10-01 09:58:19.345119: Failed to perform Auto-login: ManagedIdentityCredential authentication failed
2023-10-01 09:58:19.345131: GET http://169.254.169.254/metadata/identity/oauth2/token
2023-10-01 09:58:19.345144: --------------------------------------------------------------------------------
2023-10-01 09:58:19.345156: RESPONSE 400 Bad Request
2023-10-01 09:58:19.345168: --------------------------------------------------------------------------------
2023-10-01 09:58:19.345180: Failed to make request: '\''Identity not found: '\''ACI_MI_DEFAULT'\'''\''
2023-10-01 09:58:19.345192: 
2023-10-01 09:58:19.345204: --------------------------------------------------------------------------------
2023-10-01 09:58:19.345215: To troubleshoot, visit https://aka.ms/azsdk/go/identity/troubleshoot#managed-id.'
```

### How can we reproduce the problem in the simplest way?
Use any bash shell with User assigned managed identity, in my case it was in ev2

### Have you found a mitigation/solution?
Use client ID instead of resource string
```
export AZCOPY_MSI_CLIENT_ID="$clientId"
```

**Comment by diodfr** [comment]
The option has been deprecated :
https://github.com/Azure/azure-storage-azcopy/releases/tag/v10.21.0-Preview
> 4. Deprecated support for object IDs in MSI. Client ID or Resource ID can be used as an alternative.

**Comment by diodfr** [comment]
Please forgive me, I misread your issue.
I thought that you use the deprecated system variable.
In fact, I have the same issue #2398 but azcopy uses the system identity as fallback instead of the provided resource string.

**Comment by gapra-msft** [comment]
Hi, I believe this issue was fixed by the following PR https://github.com/Azure/azure-storage-azcopy/pull/2406

This should be released as part of our 10.22 release

**Comment by gapra-msft** [comment]
The 10.22.0 version was just released. Closing this issue as a result.
